app/controllers/proposal/collection_draft_proposals_controller.rb

```diff
@@ -95,13 +99,48 @@ def progress
 
       @fourth_information = get_progress_message('done') if get_resource.proposal_status == 'done'
 
-      @available_actions =  if get_resource.in_work?
-                              'Make additional changes or submit this proposal for approval.'
-                            elsif get_resource.submitted? || get_resource.rejected?
-                              'You may rescind this proposal to make additional changes.'
-                            else
-                              'No actions are possible.'
-                            end
+      @available_actions =  get_available_actions_text
+    end
+
+    def approve
```

Comment:
I know that Charles was moving around where the various actions were such that workflow actions are at the top, but I placed this one at the bottom and figured that it could be moved in the eventual and inevitable reconciliation of our tickets.